File: Wdtc/src/main/java/org/wdt/WdtcUI/AppMain.java
package org.wdt.WdtcUI;

import javafx.application.Application;
import javafx.application.Platform;
import javafx.scene.image.Image;
import javafx.stage.Stage;
import org.apache.commons.io.FileUtils;
import org.apache.log4j.Logger;
import org.wdt.WdtcLauncher.FilePath;

import java.io.File;
import java.io.IOException;
import java.io.InputStream;
import java.net.MalformedURLException;
import java.net.URL;

public class AppMain extends Application {
    private static final File v_m = new File(FilePath.getVersionManifestJson());
    private static final Logger logmaker = Logger.getLogger(AppMain.class);

    public static void main(String[] args) {
        launch(args);
    }

    @Override
    public void start(Stage MainStage) {
        try {
            try {
                URL url = new URL("https://www.bilibili.com");
                try {
                    InputStream in = url.openStream();
                    in.close();
                    MainStage.setTitle("Wdtc - Demo");
                } catch (IOException e) {
                    MainStage.setTitle("Wdtc - Demo (无网络)");
                    logmaker.error("* 当前无网络连接,下载功能无法正常使用!");
                }
            } catch (MalformedURLException e) {
                ErrorWin.setErrorWin(e);
            }
            Platform.runLater(() -> {
                MainStage.setWidth(615);
                MainStage.setHeight(440);
                MainStage.getIcons().add(new Image("ico.jpg"));
                MainStage.setResizable(false);
                HomeWin.setHome(MainStage);
                MainStage.show();
            });
            MainStage.setOnCloseRequest(windowEvent -> {
                if (v_m.exists()) {
                    try {
                        FileUtils.delete(v_m);
                    } catch (IOException | RuntimeException e) {
                        ErrorWin.setErrorWin(e);
                    }
                }
                logmaker.info("* 程序已退出");
                System.exit(0);
            });
        } catch (Exception e) {
            ErrorWin.setErrorWin(e);
        }
    }
}

File: Wdtc/src/main/kotlin/org/wdt/wdtc/ui/AppMain.kt
package org.wdt.wdtc.ui

import javafx.application.Application
import javafx.application.Platform
import javafx.scene.image.Image
import javafx.stage.Stage
import kotlinx.coroutines.launch
import org.wdt.wdtc.core.manger.changeSettingToFile
import org.wdt.wdtc.core.manger.currentSetting
import org.wdt.wdtc.core.manger.isDebug
import org.wdt.wdtc.core.utils.info
import org.wdt.wdtc.core.utils.isOnline
import org.wdt.wdtc.core.utils.logmaker
import org.wdt.wdtc.ui.window.*
import java.io.IOException

class AppMain : Application() {
	override fun start(mainStage: Stage) {
		if (!isOnline) logmaker.warning("当前无网络连接,下载功能无法正常使用!")
		mainStage.apply {
			title = if (isOnline) windowsTitle else getWindowsTitle("无网络")
			minWidth = windowsWidht
			minHeight = windowsHeight
			val size = getSizeManger().also {
				it.setWindwosSize()
			}
			icons.add(Image("assets/icon/ico.jpg"))
			isResizable = isDebug
			setOnCloseRequest {
				logmaker.info(size)
				try {
					currentSetting.changeSettingToFile {
						windowsWidth = width
						windowsHeight = height
					}
				} catch (e: IOException) {
					setErrorWin(e)
				}
				Platform.exit()
			}
		}.also {
			javafxScope.launch {
				HomeWindow().run { setHome(it) }
				it.show()
				logmaker.info("Window Show")
			}
		}
	}
}
